
# AI Genomics Hackathon Resources

On June 23rd, SVAI + Accel.ai + NF2 Project, are kicking off our first AI Genomics hackathon at Google Launchpad in San Francisco. This event focusses on analyzing Genomic data and research publications using AI and other computational methods. We have a special opportunity to work with patient data from an ongoing medical case, provided by Onno Faber, a serial entrepreneur and NF2 Patient.

## NF2

The main issue with NF2 is that patients develop tumors in the central nervous system (schwannoma, meningioma, ependymoma), slowly knocking out senses and body functions (hearing for most, facial paralysis and sight/muscles/sensory functions for some).

The “hallmark” of NF2 is vestibular schwannoma, a benign tumor emanating from the sheath of the VIIIth cranial nerve (vestibulocochlear nerve). NF2 patients carry one mutated NF2 allele in their germline and the second NF2 allele is inactivated somatically in the tumors following the classic Knudson’s “Two-hit hypothesis” for tumor suppressor genes. Somatic mutations of the NF2 gene are also found in sporadic schwannoma, meningioma, ependymomas as well as in other tumors types that are not typically associated with familial NF2, such as malignant mesothelioma.

## Links

- SV.AI website: [sv.ai/hackathon](https://sv.ai/hackathon/)
- Facebook: [SVAIResearch](https://www.facebook.com/groups/SVAIResearch/)
- Twitter: [#nf2hack](https://twitter.com/search?q=%23nf2hack)


## FAQ

- *Is the NF2 mutation germline?*

  The NF2 mutation in me is *not* germline, it's only found in the tumor data. The specialists think i'm mosaic.

- *Does your brother have NF2?*

  My brother does *not* have NF2. This sequence can be used for comparison.

- *How about that EGFR mutation?*

  The intronic mutation in EGFR is germline.

- *How do I redeem my gcloud credits?*

  Go to [console.cloud.google.com/billing/redeem](https://console.cloud.google.com/billing/redeem)

## Presentations

More info on the schedule on [sv.ai/s/Hackathon-Agenda](https://sv.ai/s/Hackathon-Agenda)

- *"NF2 Patient Story"* by Onno Faber, Entrepreneur

  [YouTube at 25:00](https://www.youtube.com/watch?v=uTjWNawl9Ec)

- *"Introduction to NF2"* by Marco Giovannini, MD, PhD

  [YouTube at 55:00](https://www.youtube.com/watch?v=uTjWNawl9Ec)
  ([slides](https://storage.googleapis.com/nf2project-data-fdrzgrs6tzwtdubwahwmuxtj/docs/Presentation-Giovannini-HackthonOnnoJune17.pdf))

- *"Identifying quality research papers"* by Anita Brede, CEO @ Iris.AI

  [YouTube at 01:38](https://www.youtube.com/watch?v=uTjWNawl9Ec)
  ([slides](https://www.dropbox.com/s/a0kydk6bwf6yrh5/Iris.ai%20NF2%20Genomics%20Hackathon%20June%2023-25%202017.pdf?dl=0))

  Access to the demo: [nextrc.iris.ai/demo](http://nextrc.iris.ai/demo)

  Log in to [nf2project.com/hackathon](https://nf2project.com/hackathon) for more info.

- *"AI for Drug Discovery"* by Berton Earnshaw, Dir. Data Science Research @ Recursion Pharmaceuticals

   [YouTube at 02:00](https://www.youtube.com/watch?v=uTjWNawl9Ec)

- *"Leveraging large public data for individualized medicine"* by Ben Busby, Lead, Bioinformatics Training at National Center for Biotechnology Information (NCBI)

   [YouTube at 02:18](https://www.youtube.com/watch?v=uTjWNawl9Ec)
   ([slides](https://www.slideshare.net/benbusby/leveraging-large-publicdataforindividualizedmedicine))

   - [github.com/NCBI-Hackathons](https://github.com/NCBI-Hackathons)
   - [https://ncbiinsights.ncbi.nlm.nih.gov/2017/06/15/refseq-functional-elements/](https://ncbiinsights.ncbi.nlm.nih.gov/2017/06/15/refseq-functional-elements/)
   - [https://www.ncbi.nlm.nih.gov/projects/genome/guide/human/](https://www.ncbi.nlm.nih.gov/projects/genome/guide/human/)
   - [ftp://ftp.ncbi.nih.gov/refseq/H_sapiens/RefSeqGene/](ftp://ftp.ncbi.nih.gov/refseq/H_sapiens/RefSeqGene/)
   - [GRCh38_latest_rna.fna.gz](ftp://ftp.ncbi.nlm.nih.gov/refseq/H_sapiens/annotation/GRCh38_latest/refseq_identifiers/GRCh38_latest_rna.fna.gz)


- *"Deep Learning for Drug Discovery Research"* by Bharath Ramsundar @ deepchem

  [YouTube](https://www.youtube.com/watch?v=Kuu37VyOI8I)
  ([slides](https://docs.google.com/presentation/d/1aCBemCp8XTyTifp-LzD_-aX2Qhsyg0XPL44KUCm8Q0k/edit?usp=sharing))

- *"Genomics Workshop"* by Mo Rahman, Bioinformatics & Pharmacogenomics CSO @ GeneDrop

  [YouTube](https://www.youtube.com/watch?v=K1lnhQIWOb8)
  ([slides](https://svai.slack.com/files/sean_greenslade/F5ZSNG3T8/mo_nf2hktalk.pdf))

## Research

We encourage everyone to do their own research. To help you started, here are some papers that might be helpful. Also consider using the [Iris.AI tool](http://nextrc.iris.ai/demo).

- *Current status and recommendations for biomarkers and biobanking in neurofibromatosis*

  > Clinically validated biomarkers for neurofibromatosis 1 (NF1), neurofibromatosis 2 (NF2), and schwannomatosis (SWN) have not been identified to date. The biomarker working group's goals are to (1) define biomarker needs in NF1, NF2, and SWN; (2) summarize existing data on biomarkers in NF1, NF2, and SWN; (3) outline recommendations for sample collection and biomarker development; and (4) standardize sample collection and methodology protocols where possible to promote comparison between studies by publishing standard operating procedures (SOPs).

  [http://www.neurology.org/content/87/7_Supplement_1/S40.full](http://www.neurology.org/content/87/7_Supplement_1/S40.full)


- *Genetic landscape of sporadic vestibular schwannoma*

  > Vestibular schwannoma (VS) is a benign tumor with associated morbidities and reduced quality of life. Except for mutations in NF2, the genetic landscape of VS remains to be elucidated. Little is known about the effect of Gamma Knife radiosurgery (GKRS) on the VS genome. The aim of this study was to characterize mutations occurring in this tumor to identify new genes and signaling pathways important for the development of VS. [...]

  [https://www.ncbi.nlm.nih.gov/pubmed/28409725](https://www.ncbi.nlm.nih.gov/pubmed/28409725)

- *The genomic landscape of schwannoma*

  > Schwannomas are common peripheral nerve sheath tumors that can cause debilitating morbidities. We performed an integrative analysis to determine genomic aberrations common to sporadic schwannomas. [...]

  [https://www.ncbi.nlm.nih.gov/pubmed/27723760](https://www.ncbi.nlm.nih.gov/pubmed/27723760)

- *Exploring the ‘‘Two-Hit Hypothesis’’ In NF2: Tests of Two-hit and Three-hit Models of Vestibular Schwannoma Development*
  > Neurofibromatosis 2 (NF2) is a genetic disease that occurs in approximately 1 in 40,000 live births. Almost all af fected individuals develop bilateral tumors of Schwann cells that surround the vestibular nerves; these tumors are known as vestibular schwannomas (VS). Evidence from molecular genetic studies suggests that at least two mutations are involved in formation of VS in patients with NF2. [...]

  [https://www.ncbi.nlm.nih.gov/pubmed/12687643](https://www.ncbi.nlm.nih.gov/pubmed/12687643)

# Problem Tracks

The main issue with NF2 is that patients develop tumors in the central nervous system (schwannoma, meningioma, ependymoma), slowly knocking out senses and body functions (hearing for most, facial paralysis and sight/muscles/sensory functions for some).

The “hallmark” of NF2 is vestibular schwannoma, a benign tumor emanating from the sheath of the VIIIth cranial nerve (vestibulocochlear nerve). NF2 patients carry one mutated NF2 allele in their germline and the second NF2 allele is inactivated somatically in the tumors following the classic Knudson’s “Two-hit hypothesis” for tumor suppressor genes. Somatic mutations of the NF2 gene are also found in sporadic schwannoma, meningioma, ependymomas as well as in other tumors types that are not typically associated with familial NF2, such as malignant mesothelioma.

All the ‘tracks’ below are ways to find out why these tumors are there, what makes them grow and most importantly how to stop them.

- *Is NF2 solely driven by a mutation in the “NF2 gene”, or is there more?*

  Is inactivation of NF2 gene the only reason for people to develop schwannomas, or might there be other gene alterations that influence tumor behaviour? What other affected genes could play a role in the formation and growth of these tumors?

- *Is the EGFR mutation in this tumor genome relevant for tumor formation and growth?*

  We’ve identified a somatic mutation in the EGFR gene that may play a role in the tumor development. Can you validate or invalidate that hypothesis? How?

- *What other hypotheses can we generate for why NF2-related tumors start forming and growing?*

  Are there any drugs that have not been looked at yet that may help mitigate tumor growth (like CBD)? What could be the reason for the tumor to start growing? What could be the reasons for the tumors to grow at different rates?

- *Which drugs have been studied already and what can we learn from that?*

  Which drugs have been studied so far for NF2-related tumors, and how do they rank up? Are there any drugs out there that show potential and are worth investigating more? Is there a potential gene association with this drug we may look into? What genes are targeted by any one of those drugs, and how do they affect the genetic pathways of the NF2 gene?

- *What other data may be interesting to learn from?*

  What kind of data would we need for people with NF2 in order to find out what drives tumor growth and to devise personalized treatments? Tumors, Blood, Whole genome, Proteomics, Metabolomics, -Omics, microbiome, environmental effects, lifestyle, diet.

## Dataset

The genome dataset consists of a tumor/normal pair from a patient diagnosed with the rare genetic condition NF2 and a normal from a sibling of the patient. All three are Whole Genome Sequences with a read-depth of 60X-90X and will be made available to you as .bam and .vcf files

### VCF Files

Log in to [nf2project.com/hackathon](https://nf2project.com/hackathon) to get access.

More info:

* BAM and VCF Specifications: [github.com/samtools/hts-specs](https://github.com/samtools/hts-specs)
* [vcftools.sourceforge.net/perl_module.html#vcf-isec](http://vcftools.sourceforge.net/perl_module.html#vcf-isec)

### BAM Files

The BAM files are currently not available for direct download due to the large file size. **Please use the instructions for spinning up a Google Cloud instance and the disks with the BAM files on them.** Let us know if you need to download the .bam files and we'll try to accomodate you.

The files are aligned with **hg19**. We've provided *Homo_sapiens_assembly19.fasta* on the Google Cloud and AWS images as well.

* OF_010116NF2_a (207.66 GB) has tumor data "a" from patient
* OF_112015SJIA_2 (207.19 GB) has blood/normal from patient
* NF2_XY_s (253.25 GB) has blood/normal from sibling


### gcloud Image

To get started with the gcloud image:

1. Make sure you've acquired permissions

   Go to [nf2project.com](https://nf2project.com) and make sure you've gotten permissions to the project on the Google Cloud.
If you don't have a Google Cloud account yet, you can create one at [cloud.google.com](https://cloud.google.com)


2. Create and set your own project

   First, create your project at [https://console.cloud.google.com]([https://console.cloud.google.com]).

   Note down your Google Cloud project id. You need to install and setup gcloud on the command line to continue.

	    brew cask install google-cloud-sdk
   -or-

	    https://cloud.google.com/sdk/downloads

   And then

	    gcloud auth login
	    -or-
	    https://cloud.google.com/sdk/docs/initializing


3. Now set the default project and compute zone, and make sure it's to your liking

	    gcloud config set project [GCLOUD_PROJECT_ID]
	    gcloud config set compute/zone us-west1-c
	    gcloud config list


4. Make sure your quota is sufficient for SSD disks (the disks are about 1TB in size, excluding the working disks you might need)

    [https://console.cloud.google.com/iam-admin/quotas](https://console.cloud.google.com/iam-admin/quotas)


5. Check if you have access to the images

   You should have 5 images in total. 1 for the virtual machine, 1 for the VCF files (smaller) and three seperate disks for the BAM files (about 250 GB for each file).

	    gcloud compute images list --project project-x-34f81 | grep nf2project

	    nf2project-vm-image
	    nf2project-data-vcf-image
	    nf2project-data-bam-normal-image
	    nf2project-data-bam-tumor-image
	    nf2project-data-bam-sibling-image


6. Spin up a virtual machine

   You may do this on your own, or use our Debian image (1 CPU).

        gcloud compute instances create nf2project-vm --image nf2project-vm-image --image-project project-x-34f81


7. Create the disks from the images

	    gcloud compute disks create nf2project-data-vcf --image nf2project-data-vcf-image --image-project project-x-34f81
	    gcloud compute disks create nf2project-data-bam-normal --image nf2project-data-bam-normal-image --image-project project-x-34f81
	    gcloud compute disks create nf2project-data-bam-tumor --image nf2project-data-bam-tumor-image --image-project project-x-34f81
	    gcloud compute disks create nf2project-data-bam-sibling --image nf2project-data-bam-sibling-image --image-project project-x-34f81


8. Attach the disks to the VM instance

	    gcloud compute instances attach-disk nf2project-vm --disk=nf2project-data-vcf
	    gcloud compute instances attach-disk nf2project-vm --disk=nf2project-data-bam-normal
	    gcloud compute instances attach-disk nf2project-vm --disk=nf2project-data-bam-tumor
	    gcloud compute instances attach-disk nf2project-vm --disk=nf2project-data-bam-sibling


9. SSH to the VM instance from here

   You can switch to root user for convenience.

	    gcloud compute ssh nf2project-vm

	    sudo su

10. Mount the disks on the VM instance

	    mkdir -p /mnt/disks && cd /mnt/disks

	    lsblk

	    nf2project-data-vcf (50GB)
	    nf2project-data-bam-normal (300GB)
	    nf2project-data-bam-tumor (301GB)
	    nf2project-data-bam-sibling (302GB)

	    mkdir ./data-vcf && mount /dev/[DEVICE_ID] ./data-vcf
	    mkdir ./data-bam-normal && mount /dev/[DEVICE_ID] ./data-bam-normal
	    mkdir ./data-bam-tumor && mount /dev/[DEVICE_ID] ./data-bam-tumor
	    mkdir ./data-bam-sibling && mount /dev/[DEVICE_ID] ./data-bam-sibling

11. Check out some more resources (thanks to @heuermh)

	* [reads](https://cloud.google.com/genomics/reference/rest/v1/reads) and [variants](https://cloud.google.com/genomics/reference/rest/v1/variants) from the Google Genomics API
	* [reads](http://ga4gh-schemas.readthedocs.io/en/latest/api/index.html#reads) and [variants](http://ga4gh-schemas.readthedocs.io/en/latest/api/variants.html) from the GA4GH APIs
	* [alignment records](https://github.com/bigdatagenomics/bdg-formats/blob/master/src/main/resources/avro/bdg.avdl#L136) and [variants](https://github.com/bigdatagenomics/bdg-formats/blob/master/src/main/resources/avro/bdg.avdl#L730) in [ADAM](https://github.com/bigdatagenomics/adam)